### PR TITLE
Add retry logic for Django and Psycopg2 InterfaceErrors

### PR DESCRIPTION
**Pull request by juspence**
This PR implements https://github.com/celery/django-celery-results/issues/119 in a slightly different way. It adds retry logic only for django.db.utils.InterfaceError and psycopg2.InterfaceError, which can be errors like "connection already closed".

It does not retry any DatabaseError, which can be errors like "numeric value out of range" or "foreign key check failed". See https://www.psycopg.org/docs/module.html#exceptions for more info.

exception_safe_to_retry() is overridden from the parent class: https://docs.celeryproject.org/en/stable/_modules/celery/backends/base.html#BaseBackend

This method is called when an exception is raised in _store_result() and always_retry / result_backend_always_retry is set to True. If the exception is an InterfaceError, the connection is closed as a side effect of calling exception_safe_to_retry().

Then the retry logic implemented in the parent class continues, and Django will automatically open a new connection for the next attempt to store the task result or get the task metadata.

### CODE REVIEW

**Review by auvipy — CHANGES_REQUESTED** [review]
I would like to see unit tests for this change

**Comment by auvipy** [comment]
I want to add this to next release, can you please address the review comment?

**Comment by juspence** [comment]
@auvipy Yes, I am working on this today since I had other tasks to finish Friday.

**Review by auvipy — APPROVED** [review]

File: django_celery_results/backends/database.py [hunk 1]
```diff
@@ -7,13 +7,17 @@
 from celery.result import GroupResult, allow_join_result, result_from_tuple
 from celery.utils.log import get_logger
 from celery.utils.serialization import b64decode, b64encode
-from django.db import transaction
+from django.db import connection, transaction
+from django.db.utils import InterfaceError
 from kombu.exceptions import DecodeError
+from psycopg2 import InterfaceError as Psycopg2InterfaceError
```
Comment (intgr) [inline django_celery_results/backends/database.py:13]: This adds a hard dependency on psycopg2, but django-celery-results can be used with other databases as well.

I have opened PR #292 to fix this.

**Pull request merged**